### ISSUE TITLE
Add ARIA roles and attributes to the modal dialog component

### ISSUE THREAD

**Issue by Veyfeyken** [a-ui/core_branding_scss#157]
### Problem

[molecules/molecules.modal](https://a-ui.github.io/core_branding_scss/#molecules-modal)

* The markup of the modal dialog is missing ARIA in order to work properly with assitive technologies such as screenreaders.
* A modal dialog forces the user to take action and should interrupt/hide the active page. A non-modal dialog does not. Since this is a modal dialog a close button shouldn't be available?
* Focus management is essential for a modal dialog to be accessible. Add a note to the documentation?

![screenshot modal](https://user-images.githubusercontent.com/870049/98671712-f172f500-2354-11eb-8431-488eda1a16bf.jpg)

### Solution

* Follow the ARIA design pattern for modal dialog.
* Delete the close button?

### Articles and links

[ARIA design pattern: Dialog (Modal)](https://www.w3.org/TR/wai-aria-practices-1.2/#dialog_modal)

**Comment by TriangleJuice** [comment]
Hey @Veyfeyken this issue should be fixed in the next iteration of our core branding (`v4.3.0`)
